libs/template/src/lib/core/utils/template-manager_creator.ts

```diff
@@ -0,0 +1,79 @@
+import { EmbeddedViewRef, TemplateRef, ViewContainerRef } from '@angular/core';
+
+type RxTemplateName = 'rxNext' | 'rxComplete' | 'rxError' | 'rxSuspense';
+
+export interface TemplateManager<T> {
+  updateViewContext(viewContextSlice: Partial<T>): void;
+  addTemplateRef(name: RxTemplateName, templateRef: TemplateRef<T>): void;
+  insertEmbeddedView(name: RxTemplateName): void;
+  destroy(): void;
+}
+
+/**
+ * TemplateManager
+ *
+ * @description
+ * This function returns an object that holds the logic for managing templates to
+ * be embedded into a `ViewContainerRef`. Internally it caches template references and instantiates
+ * embedded views. The `TemplateManager` lets you re-use templates and insert views on-demand (e.g.
+ * when a new observable notification is sent) and update the view context.
+ *
+ * @param viewContainerRef reference to a top-level view container templates will be attached to
+ * @param initialViewContext initial view context state
```

Comment:
```suggestion
 * This function returns an object that holds the logic for managing templates of a `ViewContainerRef`. 
 * It abstracts `EmbeddedView` creation, `insert` calls and `ViewContext` updates.
 * Internally the creations of template references are lazy, and a combination of caching logic and the `ViewContainerRef#detach` method.
 * The `TemplateManager` lets you re-use templates and insert views on-demand, as well as update the view context.
 * (e.g. when a new observable notification is sent) 
 *
 * @param viewContainerRef reference to a top-level view container where the passed templates will be attached to
 * @param initialViewContext initial view context state
```

Reply:
Updated the jsdoc and applied the suggestion with some minor tweaks ;)